### PR TITLE
Refactor remote client execution model

### PR DESCRIPTION
**Pull request by cmin764**
- reuse the code and eliminate redundancy
- extend the retrying capability to its roots
- access to custom methods directly from the `remote_client` utility object

### CODE REVIEW

File: argus/util.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import six
 
 from argus import config
+from argus import exceptions as errors
```
Comment (PCManticore) [inline argus/util.py]: I prefer to use exceptions instead.

File: argus/util.py [hunk 2]
```diff
@@ -43,17 +44,92 @@
 
 
 class WinRemoteClient(remote_client.WinRemoteClient):
-    def run_command_verbose(self, cmd):
-        """Run the given command and log anything it returns."""
+
+    def run_command_verbose(self, cmd, print_all=True):
```
Comment (PCManticore) [inline argus/util.py]: I don't like how print_all works.

Comment (cmin764) [inline argus/util.py]: Fixed in the newest patch.

File: argus/util.py (hunk 1 above)
Comment (cmin764) [inline argus/util.py]: Was of run_once decorator.

**Pull request merged**